util/topgen/templates/plic_all_irqs_test.c.tpl

```diff
@@ -112,13 +115,14 @@ void ottf_external_isr(uint32_t *exc_info) {
 
   top_${top["name"]}_plic_peripheral_t peripheral = (top_${top["name"]}_plic_peripheral_t)
       top_${top["name"]}_plic_interrupt_for_peripheral[plic_irq_id];
+% if has_csrng:
   // TODO(lowrisc/opentitan#20747) Adjust code once this issue is fixed.
   if (allow_csrng_irq && kBootStage == kBootStageOwner &&
       peripheral != peripheral_expected &&
-      peripheral == kTopEarlgreyPlicPeripheralCsrng) {
+      peripheral == kTop${top["name"].capitalize()}PlicPeripheralCsrng) {
```

Comment:
I wonder whether it's worth defining a variable for the "namespace" like
```
  k_plic_pfx = 'kTop{}Plic'.format(top['name'].capitalize())
```
(untested!)

Then this line could look more like
```
  peripheral == ${k_plic_pfx}PeripheralCsrng
```
and similarly below.

No big stress, but it might be slightly easier to follow?

Reply:
I will pass on your suggestion since this code may be removed once the TODO is addressed.